src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -74,15 +74,46 @@ static ucs_status_t uct_cuda_ipc_iface_get_address(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+#if HAVE_CUDA_FABRIC
+static int uct_cuda_ipc_iface_is_coherent()
+{
+    CUdevice cu_device;
+    int coherent;
+    ucs_status_t status;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&cu_device, 0));
+    if (status != UCS_OK) {
+        return 0;
+    }
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(
+            cuDeviceGetAttribute(&coherent,
+                                 CU_DEVICE_ATTRIBUTE_PAGEABLE_MEMORY_ACCESS_USES_HOST_PAGE_TABLES,
```

Comment:
is this an indirect indication that MNNVL is supported?
is there a more direct check? like trying to allocate fabric memory?

Reply:
This is an indirect check. Fabric memory allocation doesn't suffice. 

On non-MNNVL capable systems which have hardware coherence, the checks [here](https://github.com/openucx/ucx/pull/9982/files#diff-561a7e4fa208245423bb161bc7a1cc05fbce0d65a10ef13c77e366f38c541826R231) will suffice. This is where we require protov2 to fail gracefully and fallback to IB-based rendezvous transfers between nodes. 

This check was added mainly to avoid attempting MNNVL on non-coherent systems.